diff --git a/dist/codegen/createBarrelFiles.js b/dist/codegen/createBarrelFiles.js
index 66c455838b7d5b4fb7343e6b37aad2777d0baa2d..863c7a9578f86e41fdfdf507f2de6a07a776e613 100644
--- a/dist/codegen/createBarrelFiles.js
+++ b/dist/codegen/createBarrelFiles.js
@@ -36,7 +36,13 @@ function createBarrelFiles(paths, { typeOnly, postfix = '', moduleSuffix = '' })
         const nestedDirs = (directoryReexports[path] || []).sort();
         const namespacesExports = nestedDirs
             .map((p) => {
-            const namespaceIdentifier = (0, lodash_1.camelCase)(p);
+            let namespaceIdentifier = (0, lodash_1.camelCase)(p);
+
+            // local patch: prepend underscore if directory has a leading digit
+            if (/^\d/.test(namespaceIdentifier)) {
+                namespaceIdentifier = "_" + namespaceIdentifier
+            }
+            
             if (typeOnly)
                 return [
                     `import type * as ${namespaceIdentifier} from './${p}';`,
